menuinst/platforms/base.py

```diff
@@ -137,6 +140,14 @@ def __init__(self, menu: Menu, metadata: dict):
         self.menu = menu
         self._data = self._initialize_on_defaults(metadata)
         self.metadata = self._flatten_for_platform(self._data)
+        if isinstance(self.metadata["name"], dict):
+            if self.menu.env_name:
+                name = self.metadata["name"].get("target_environment_is_not_base", "")
+            else:
+                name = self.metadata["name"].get("target_environment_is_base", "")
+            if not name:
+                raise KeyError("Cannot parse `name` from dictionary representation.")
+            self.metadata["name"] = name
```

Comment:
At this point we have not rendered the `{{ variables }}`, right?

Reply:
That's correct.

The tests for a dictionary representation of the name would have failed by then, so even if code changes render the name at that point, the tests will catch that.